subsys/net/ip/net_if.c

```diff
@@ -356,6 +356,12 @@ enum net_verdict net_if_send_data(struct net_if *iface, struct net_pkt *pkt)
 		goto done;
 	}
 
+	if (unlikely(!net_if_l2(iface))) {
```

Comment:
if (IS_ENABLED(CONFIG_NET_OFFLOAD) && unliely()...

Reply:
@tbursztyka ok, assuming CONFIG_NET_OFFLOAD would be the only reason l2 is NULL

Reply:
Yes, that's the only case so far